cypress-tests/cypress/e2e/v2/anonymousNavigation.cy.ts

```diff
@@ -60,6 +61,9 @@ describe("Anonymous users can only access public resources", () => {
         });
       }
     });
+
+    // ? Wait a little time to allow the search engine to index resources
+    cy.wait(TIMEOUTS.short);
```

Comment:
why not query the search API with a loop and timeout here? this will prevent flakes and avoid guessing the indexing time.

Feel free to ignore in this PR, but then track this in an issue.

Reply:
the other solution is to repeatedly search with the UI (click again on the search button) until we get the expected result.

Reply:
I like your first suggestion! Implemented here https://github.com/SwissDataScienceCenter/renku/pull/3906/commits/f1814398bec0ffc5eb4aa430d2f3752492c58df2